lib/services/ngsi/entities-NGSI-v2.js

```diff
@@ -323,6 +324,7 @@ function sendQueryValueNgsi2(entityName, attributes, typeInformation, token, cal
  */
 function sendUpdateValueNgsi2(entityName, attributes, typeInformation, token, callback) {
     let payload = {};
+    const explicitAttrsList = ['TimeInstant'];
```

Comment:
Given the change on TimeStamp behavior, a test that cover and checks JEXL expressions + explicitAttrs + autogenerated Timestamp and provisioned Timestamp should be included

Reply:
Fixed 39c4744 - NGSI-v2 and NGSI-LD tests added.

Reply:
Correct me if I am wrong, with the changes included in this PR, if you send a measure, `TimeInstant:foo`, using the flag `timestamp:false`, this measure would progress to the entity because of declaring `const explicitAttrsList = ['TimeInstant'];`

Reply:
The [relevant code](https://github.com/telefonicaid/iotagent-ul/blob/master/lib/commonBindings.js#L127) in the original can be found in `manageConfigurationRequest(apiKey, deviceId, device, objMessage)` - which is due to be deleted. If you supply `TimeInstant' it is propagated through to the context broker.

Incidentally this is where the original _"Magic Number"_ was taken from.

Reply:
As in the example above, you can generate a `TimeInstant` from an expression and supply a unix timestamp rather than an ISO date too:

```json
"attributes": [
                { "name": "TimeInstant", "type": "DateTime", "expression": "ts|toisodate" }
]
```

```console
curl -L -X POST 'http://localhost:7896/iot/json?k=<apikey>&i=<deviceId>' \
-H 'Content-Type: application/json' \
--data-raw '{"latitude": 3, "longitude": 20, "level": 30, "ts" : 1}'
```

Reply:
What I am highlighting here is different.

If you provision an entity like this (note the option [`"timestamp":false`](https://github.com/telefonicaid/iotagent-node-lib/blob/master/doc/api.md#device-model)):

```JSON
...
 "timestamp":false
 "active": [
     { "name": "attr",  "type": "number" }
 ],
 "explicitAttrs": true
...
```

And you send a measure like this one:

```JSON
 {"TimeInstant": "foo"}
```

The resulting resulting update operation would include an attribute `TimeInstant` even if it was not defined in the provision and `explicitAttrs` value was `true`

Reply:
Fixed 710fabe - test case added.